lib/cyclone-core/src/progress.rs

```diff
@@ -146,9 +144,20 @@ impl FunctionResultFailure {
     }
 }
 
+#[remain::sorted]
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone, Display)]
+pub enum FunctionResultFailureErrorKind {
+    ActionFieldWrongType,
+    CancelledExecution,
+    InvalidReturnType,
+    ReconciliationFieldWrongType,
+    UserCodeException(String),
+    VeritechServer,
+}
+
 #[derive(Debug, Deserialize, Eq, PartialEq, Serialize, Clone)]
 pub struct FunctionResultFailureError {
-    pub kind: String,
+    pub kind: FunctionResultFailureErrorKind,
```

Comment:
This was the scariest part to add, since it's carried all the way from cyclone back to the DAL basically. From what I've tested it does not break any protocols but I'd appreciate extra eyes

Reply:
Looks good to me. You also don't have to worry about #4438 because we only dropped `Reconciliation` funcs from veritech (still "above" in dal, still "below" in cyclone and lang-js).